pkg/query/remote_engine.go

```diff
@@ -246,10 +247,18 @@ func (r *remoteQuery) Exec(ctx context.Context) *promql.Result {
 				F: s.Value,
 			})
 		}
-		for _, h := range ts.Histograms {
+		for _, hp := range ts.Histograms {
+			var fh *histogram.FloatHistogram
+
+			if hp.IsFloatHistogram() {
```

Comment:
AFAIK we can only get back float histograms from the engine, so maybe this check is not needed.

Reply:
You are right. Bot vector and matrix only have float histogram. I remove it.